log/sequencer.go

```diff
@@ -210,3 +246,60 @@ func (s Sequencer) SequenceBatch(limit int) (int, error) {
 
 	return len(leaves), nil
 }
+
+// SignRoot wraps up all the operations for creating a new log signed root.
+func (s Sequencer) SignRoot() error {
+	tx, err := s.logStorage.Begin()
+
+	if err != nil {
+		glog.Warningf("signer failed to start tx: %s", err)
+		return err
+	}
+
+	// Get the latest known root from storage
+	currentRoot, err := tx.LatestSignedLogRoot()
+
+	if err != nil {
+		glog.Warningf("signer failed to get latest root: %s", err)
+		tx.Rollback()
+		return err
+	}
+
+	// Initialize a Merkle Tree from the state in storage. This should fail if the tree is
+	// in a corrupt state.
+	merkleTree, err := s.initMerkleTreeFromStorage(currentRoot, tx)
+
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	// Build the updated root, ready for signing
+	newLogRoot := trillian.SignedLogRoot{
+		RootHash:       merkleTree.CurrentRoot(),
+		TimestampNanos: s.timeSource.Now().UnixNano(),
+		TreeSize:       merkleTree.Size(),
+		LogId:          currentRoot.LogId,
+		TreeRevision:   currentRoot.TreeRevision + 1,
```

Comment:
I think it'd make sense for the sequencer process to store the signed tree root, rather than having a separate signer process - having the two things do similar work and having both re-calculate the `TreeRevision` scares me a bit.